e2e/_suites/ingest-manager/fleet.go

```diff
@@ -299,109 +338,428 @@ func (fts *FleetTestSuite) systemPackageDashboardsAreListedInFleet() error {
+			"hostname":      fts.Hostname,
+			"matches":       matches,
+			"retries":       retryCount,
+		}).Info("The Agent is listed in the Administration view in the Security App in the desired status")
+		return nil
+	}
+
+	err := backoff.Retry(agentListedInSecurityFn, exp)
 	if err != nil {
 		return err
 	}
 
 	return nil
 }
 
-func (fts *FleetTestSuite) theEnrollmentTokenIsRevoked() error {
-	log.WithFields(log.Fields{
-		"token":   fts.CurrentToken,
-		"tokenID": fts.CurrentTokenID,
-	}).Debug("Revoking enrollment token")
+func (fts *FleetTestSuite) anEndpointIsSuccessfullyDeployedWithAgent(image string) error {
+	err := fts.anAgentIsDeployedToFleet(image)
+	if err != nil {
+		return err
+	}
 
-	err := fts.removeToken()
+	err = fts.theAgentIsListedInFleetWithStatus("online")
+	if err != nil {
+		return err
+	}
+
+	// we use integration's title
+	return fts.theIntegrationIsOperatedInThePolicy("Elastic Endpoint", actionADDED, "default")
+}
+
+func (fts *FleetTestSuite) thePolicyResponseWillBeShownInTheSecurityApp() error {
+	agentID, err := getAgentID(fts.Hostname)
+	if err != nil {
+		return err
+	}
+
+	maxTimeout := 2 * time.Minute
+	retryCount := 1
+
+	exp := e2e.GetExponentialBackOff(maxTimeout)
+
+	getEventsFn := func() error {
+		listed, err := isPolicyResponseListedInSecurityApp(agentID)
+		if err != nil {
+			log.WithFields(log.Fields{
+				"elapsedTime": exp.GetElapsedTime(),
+				"err":         err,
+				"retries":     retryCount,
+			}).Warn("Could not get metadata from the Administration view in the Security App yet")
+			retryCount++
+
+			return err
+		}
+
+		if !listed {
+			log.WithFields(log.Fields{
+				"agentID":     agentID,
+				"elapsedTime": exp.GetElapsedTime(),
+				"retries":     retryCount,
+			}).Warn("The policy response is not listed as 'success' in the Administration view in the Security App yet")
+			retryCount++
+
+			return fmt.Errorf("The policy response is not listed as 'success' in the Administration view in the Security App yet")
+		}
+
+		log.WithFields(log.Fields{
+			"elapsedTime": exp.GetElapsedTime(),
+			"retries":     retryCount,
+		}).Info("The policy response is listed as 'success' in the Administration view in the Security App")
+		return nil
+	}
+
+	err = backoff.Retry(getEventsFn, exp)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (fts *FleetTestSuite) thePolicyIsUpdatedToHaveMode(name string, mode string) error {
+	if name != "malware" {
+		log.WithFields(log.Fields{
+			"name": name,
+		}).Warn("We only support 'malware' policy to be updated")
+		return godog.ErrPending
+	}
+
+	if mode != "detect" && mode != "prevent" {
+		log.WithFields(log.Fields{
+			"name": name,
+			"mode": mode,
+		}).Warn("We only support 'detect' and 'prevent' modes")
+		return godog.ErrPending
+	}
+
+	integration, err := getIntegrationFromAgentPolicy("Elastic Endpoint", fts.PolicyID)
+	if err != nil {
+		return err
+	}
+	fts.Integration = integration
+
+	integrationJSON := fts.Integration.json
```

Comment:
We are storing the JSON representation of the integration package policy right after it's created. Then we have to remove a few fields that are not accepted by the  update API call.